### DISCUSSION TITLE
Image of the Soviet single-seat lunar lander

### DISCUSSION THREAD

**Discussion by peyre** [raceintospace/raceintospace#588] [Ideas]
I wonder if we should consider changing the image of the LK Lander.  The one we inherited from BARIS (what it called the L-3) doesn't look like the real thing and, well, I think it looks kind of lame.   Of course, that was because they really didn't know what it looked like at the time.  I totally get that, but don't see why we couldn't/shouldn't update RIS with newer imagery if we can.  Besides, the real-life LK looks so cool!

Recently I changed the image of the Atlas rocket because the one we inherited from BARIS also looked nothing like a real Atlas (and was even lamer), and that also brought it in line with what the mission movies show.  The mission movies for the LK would still look wrong, but it would look right in R&D/Purchasing.  I don't know how feasible it would ever be to replace the LK movies, given the trouble I've had trying to replace N1, LK-700, and Lapot movies.

![rdfull but 54](https://user-images.githubusercontent.com/3118014/135317899-536a2a4e-cc56-4016-a769-e91c034fcb03.png) ![LK](https://user-images.githubusercontent.com/3118014/135317920-be2620a0-5a8a-47c7-bdee-f1325bd182a6.jpg)

**Comment by hweimer** [comment]
Sure, why not? Wikimedia Commons has an LK image that has been placed in the public domain, so we could use that as a starting point. With the movies, I'm not sure. Unlike NASA footage, movies created by the Soviet space program have not been released under a free (i.e., GPL-compatible) license, so we cannot simply take what we find on the internet. Funnily, the article on the N1 on Wikipedia uses a photograph taken by a U.S. spy for the same reason.

**Comment by peyre** [comment]
Better yet - it's important to keep the same kind of color scheme etc., so it blends well with the other lander.  So rather than downsampling a photo, I've modified the LKM (Duet) to look more like the LK.  For the drawing-board image I used the Wikimedia image I found.  I think the results are good!
![liftoff abz SVC6 (copy 1)](https://user-images.githubusercontent.com/3118014/135765772-6e6c6258-9b58-4e9f-9b9e-e10ec1d68f41.png) -> ![liftoff abz SVC6](https://user-images.githubusercontent.com/3118014/135765785-78a2e305-507b-4cc1-a97b-fd2eb55bdf60.png) AND ![rdfull but 54](https://user-images.githubusercontent.com/3118014/135765773-49674459-0baf-4d24-bcac-050b5bf25ad9.png) -> ![rdfull but 54](https://user-images.githubusercontent.com/3118014/135765795-d3b04ac4-e6bd-42c7-9747-7f1dfbf888e9.png)

**Reply by hweimer** [reply]
Wow, that's so much better!

**Reply by peyre** [reply]
Thanks!!  I also updated the images in the VIB and the CIA/KGB report.

This is the change for the VIB:
![vab img 1-old](https://user-images.githubusercontent.com/3118014/135900430-b2d9fcda-908a-4365-836b-580d56763339.png) -> ![vab img 1](https://user-images.githubusercontent.com/3118014/135900432-79d8dd8d-f4d7-418b-b150-1d655d153bd2.png)

This is the change for the CIA/KGB report:
![cia but 0-old](https://user-images.githubusercontent.com/3118014/135900474-e26fd277-305d-4eaa-adec-0039c1e5c3a7.png) -> ![cia but 0](https://user-images.githubusercontent.com/3118014/135900475-78911ee8-2969-4f70-baa2-f6c935a274f9.png)

**Comment by peyre** [comment]
Uh oh, found a couple more!  I'm  not sure where that cartoony one is used though, so it's hard to test.

![lmer but 5](https://user-images.githubusercontent.com/3118014/137599110-955d1a12-7a1c-4db6-a557-4ad18536caee.png) -> ![lmer but 5](https://user-images.githubusercontent.com/3118014/137599105-d3b9c51b-b5bd-415f-bbfd-d53b4a720e98.png)


![aprog 1 7](https://user-images.githubusercontent.com/3118014/137599116-ddf58e3a-ec42-4595-8069-ae1798239ad8.png) -> ![aprog 1 7](https://user-images.githubusercontent.com/3118014/137599106-c8ed4949-9abb-4ce5-9a00-96800a3d91bb.png)

**Reply by peyre** [reply]
I think the "cartoony" images are what are used in the capsule/shuttle program building - so the Apollo one would be used in the Apollo Program building.  So images that don't have such a building (ie, anything not a shuttle or capsule) are available but not used, as far as I can tell.  Oh well, it's updated if we ever need it.

**Comment by peyre** [comment]
So, these images are working great so far, though one thing I dislike is that the new liftoff.abz.SVC6.png has a black (transparent) background, in contrast to all the other drawing-board images.  If I change the background to white, it messes with the actual image.  Can this be fixed?

**Reply by peyre** [reply]
It seems that weirdness with the background was because these are supposed to be grayscale and when the image is in color, the black background shows.  In my editing below I noticed this and changed my updated image to grayscale, and the background looked right.

**Comment by peyre** [comment]
Another thought about this: the movies still look wrong, of course, and fixing them would not be trivial.  I wonder if it would be desirable, for the time being, to simply use the LKM (Duet) videos for the LK.  They look a bit more like the real thing, though that would have the drawback that both look identical in the mission movies.  You wouldn't know which the Soviets just tested in lunar orbit for instance.  

Down the road, we might be able to put together movies of the LK using the Orbiter game.  Take the appropriate clips, downsample them to the same low res as the other mission movies and strip out the sound.  Of course, I've never had success myself in creating replacement movies for the N1, UR-700, and Lapot, so that's an extra complication.

Alternatively, we could extract the images from the existing LKM/Duet videos and modify them frame by frame, then rebuild the movies.  I'd be willing to do that if I knew how.

**Comment by peyre** [comment]
I'm pleased with what's done so far, but haven't been very happy with the drawing-board image for the LK.  It just doesn't match the style of the other equipment, especially the two-seat lander:

![LKM](https://user-images.githubusercontent.com/3118014/159960421-1c1eb52a-18af-4130-a1cc-d09e9d25dfe6.png) ![LK old](https://user-images.githubusercontent.com/3118014/159960396-4c95910b-59e5-4d83-9b26-5039f46f281d.png)

I've been tinkering with the image and have tried to make it closer.  Here are two I've put together; with the one on the right I told IrfanView to sharpen the image.  Does anyone have any thoughts about which is better?  I'm thinking the one on the right is almost approaching the style of the other lander.

![LK new](https://user-images.githubusercontent.com/3118014/159960946-2a2705a1-a109-4b93-b28b-e4fa11227bae.png) ![LK new sharpened](https://user-images.githubusercontent.com/3118014/159960965-19bfc379-c4ae-48bb-8ba0-0e02714bca70.png)

**Comment by peyre** [comment]
It occurred to me I might have lost some resolution in all my fiddling with the image (even though PNG isn't a lossy format), so I went back to the original free Wikimedia image, and came up with this.  I think it's the best of the bunch.

![LK - from scratch - cropped](https://user-images.githubusercontent.com/3118014/160873531-e7dac4f8-dde2-4cc2-b626-c9e693e06153.png)

**Comment by rnyoakum** [comment]
I agree. This image has more clarity than the others.

**Reply by peyre** [reply]
Thanks Ryan!  I'll test it this weekend and upload, if it all looks good.

Or better yet, tomorrow!  I forgot it's a holiday for us.

**Comment by peyre** [comment]
Ok, I got that image uploaded.  I just tested the LK because I've been wondering about the spinning drawing-board image in the Step Failure screen.  If it used some sophisticated algorithm to create a spinning image from the still we were in good shape, but if not, we still had something else to work on.  Sure enough it's about what I figured.  Anyone know where these are located?  I haven't been able to find them.

![Step Failure](https://user-images.githubusercontent.com/3118014/161113880-dc9987d4-7d5f-4c4f-be5e-998693d317dc.png)

**Reply by peyre** [reply]
Ugh, I see it's encoded deep within data/gamedata/liftoff.abz.  I guess this one will have to wait too.

**Reply by rnyoakum** [reply]
So, I've looked at liftoff.abz in the past. It doesn't have an algorithm to create a spinning image. It uses a series of still images referenced in an animation path to create a video. If you need specifics, or even a summary of liftoff.abz, I can provide them.

**Reply by peyre** [reply]
Really!  Sure, that would be good to know.  I know the _spinning image for an LK step failure_ is one of the most obscure parts of the game, since no one would see it but a Soviet player who was flying Voskhod or using Fritz's backup-capsule strategy.  But I'd be willing to try my hand at updating the images.

**Reply by rnyoakum** [reply]
@peyre - I've provided an explanation of liftoff.abz over [here](https://github.com/raceintospace/raceintospace/discussions/637). Sorry it was so long in coming. I had been in the process of moving the RIS code dealing with the file into a new class, so I didn't get the report finished until after I submitted 40f8022406a7236edae66cd15f0c1ef174c59ba0.

I have not checked if the pixel data is compressed or uncompressed for the relevant animation. If it's compressed, you'll need to write a program to edit the file & manage the headers. If not, a good hex editor might suffice.
